Mathlib/Order/Filter/Basic.lean

```diff
@@ -2067,6 +2070,26 @@ theorem pure_bind (a : α) (m : α → Filter β) : bind (pure a) m = m a := by
   simp only [Bind.bind, bind, map_pure, join_pure]
 #align filter.pure_bind Filter.pure_bind
 
+lemma eq_pure_iff_singleton_mem {l : Filter α} [hl : NeBot l] {x : α} :
```

Comment:
How about first proving:
```lean
lemma le_pure_iff_eq_pure {l : Filter α} [hl : NeBot l] :
    l ≤ pure x ↔ l = pure x := by
  refine' ⟨fun h ↦ _, fun h ↦ h.le⟩
  -- etc.
```
and then proving `eq_pure_iff_singleton_mem` using `le_pure_iff_eq_pure` and `le_pure_iff` (which does already exist, albeit lower down in the file).